server/sonar-web/src/main/js/app/components/nav/global/__tests__/GlobalNavSearchForm-test.js

```diff
@@ -0,0 +1,119 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+// @flow
+import React from 'react';
+import { shallow } from 'enzyme';
+import type { ShallowWrapper } from 'enzyme';
+import GlobalNavSearchForm from '../GlobalNavSearchForm';
+import { elementKeydown } from '../../../../../helpers/testUtils';
+
+function render(props?: Object) {
+  return shallow(
+    // $FlowFixMe
+    <GlobalNavSearchForm
+      appState={{ organizationsEnabled: false }}
+      currentUser={{ isLoggedIn: false }}
+      onClose={jest.fn()}
+      {...props}
+    />
+  );
+}
+
+function component(key: string, qualifier: string = 'TRK') {
+  return { key, name: key, qualifier };
+}
+
+function next(form: ShallowWrapper, expected: string) {
+  elementKeydown(form.find('input'), 40);
+  expect(form.state().selected).toBe(expected);
+}
+
+function prev(form: ShallowWrapper, expected: string) {
+  elementKeydown(form.find('input'), 38);
+  expect(form.state().selected).toBe(expected);
+}
+
+function select(form: ShallowWrapper, expected: string) {
+  // $FlowFixMe
+  form.instance().handleSelect(expected);
+  expect(form.state().selected).toBe(expected);
+}
+
+it('renders different components and dividers between them', () => {
+  const form = render();
+  form.setState({
+    results: {
+      TRK: [component('foo'), component('bar')],
+      BRC: [component('qwe', 'BRC'), component('qux', 'BRC')],
+      FIL: [component('zux', 'FIL')]
+    }
+  });
+  expect(form.find('.menu')).toMatchSnapshot();
+});
+
+it('renders "Show More" link', () => {
+  const form = render();
+  form.setState({
+    more: { TRK: 175, BRC: 0 },
+    results: {
+      TRK: [component('foo'), component('bar')],
+      BRC: [component('qwe', 'BRC'), component('qux', 'BRC')]
+    }
+  });
+  expect(form.find('.menu')).toMatchSnapshot();
+});
+
+it('selects results', () => {
+  const form = render();
+  form.setState({
+    results: {
+      TRK: [component('foo'), component('bar')],
+      BRC: [component('qwe', 'BRC')]
+    },
+    selected: 'foo'
+  });
+  expect(form.state().selected).toBe('foo');
+  next(form, 'bar');
+  next(form, 'qwe');
+  next(form, 'qwe');
+  prev(form, 'bar');
+  select(form, 'foo');
+  prev(form, 'foo');
+});
+
+it('opens selected on enter', () => {
+  const form = render();
+  form.setState({
+    results: { TRK: [component('foo')] },
+    selected: 'foo'
+  });
+  const openSelected = jest.fn();
+  // $FlowFixMe
+  form.instance().openSelected = openSelected;
+  elementKeydown(form.find('input'), 13);
+  expect(openSelected).toBeCalled();
+});
+
+it('closes on escape', () => {
```

Comment:
check this out https://github.com/SonarSource/sonarqube/pull/2030/commits/39998d11a81737a5820d444938a9fcf799c5b6e3#diff-2dd310a35ef7c548bae94baef415fadcR121